### ISSUE TITLE
torch.cross is divergent from np.cross

### ISSUE THREAD

**Issue by mruberry** [pytorch/pytorch#50273]
PyTorch's [torch.cross](https://pytorch.org/docs/master/generated/torch.cross.html?highlight=cross#torch.cross) has the  signature:

```
torch.cross(input, other, dim=None, *, out=None) → Tensor
```

And its documentation notes that "If dim is not given, it defaults to the first dimension found with the size 3. Note that this might be unexpected."

Numpy's [np.cross](https://numpy.org/doc/stable/reference/generated/numpy.cross.html) has the signature:

```
numpy.cross(a, b, axisa=-1, axisb=-1, axisc=-1, axis=None)
```

It does not "search" for a dimension of size 3, and will allow dimensions of size 2 or 3 to be used in its computation.

np.cross is a top 200 NumPy function according to a static analysis of NumPy function usage (see [here](https://github.com/Quansight-Labs/python-api-inspect/blob/master/data/csv/numpy-summary.csv)). Within Facebook there are hundreds of uses of either PyTorch's or NumPy's cross function, too.

**Comment by mruberry** [comment]
For additional context, the latest array API has a different signature for cross:

https://data-apis.org/array-api/latest/API_specification/linear_algebra_functions.html#cross-x1-x2-axis-1

**Comment by rgommers** [comment]
> For additional context, the latest array API has a different signature for cross:

I looked at the PR adding it, and the rationale was: _"The spec does not include support for length 2 vectors when computing the cross product (NumPy) or specifying axes independently (NumPy), as this is not supported by Torch or TensorFlow."_

I think `numpy.cross` is a bit of a mess, and this rationale is quite reasonable. This looks fine as a signature:
```
cross(x1, x2, /, *, axis=-1)
```
and is basically the same as `torch.cross`. Searching through code bases of SciPy, scikit-learn, scikit-image and Astropy, I only find some handfuls of 2-argument `np.cross` usage. The `axisa`, `axisb`, `axisc` thing is weird.

Also, this is a linear algebra function and I think it's too niche to be in the main namespace. There really is no optimal way to split this, and we're still puzzling on it at (https://github.com/data-apis/array-api/issues/165, https://github.com/data-apis/array-api/pull/149), but we're thinking most linear algebra functions (including `cross`) really belong in the `linalg` namespace and only a few of the most popular ones that are also relatively straightforward to implement with BLAS/LAPACK (like `transpose`, `einsum`) in the main namespace.

tl;dr I'd hold off on making any change to `torch.cross` yet, and see how this ends up in the array API standard.

**Comment by mruberry** [comment]
Thanks for the update, @rgommers. One angle to start considering linalg vs. not is whether we'd be willing to separate a function from a "core" PyTorch distribution vs. one with optional packages (like linalg and special). For operations that are linear algebra operations but also "critical," like matmul, maybe they're offered as both torch.matmul and torch.linalg.matmul, or maybe those "critical" operations are in torch and only "separable" operations are in the namespaces.

**Comment by rgommers** [comment]
> One angle to start considering linalg vs. not is whether we'd be willing to separate a function from a "core" PyTorch distribution vs. one with optional packages (like linalg and special)

Yes, I think that should be an important consideration. Also in light of package size headaches, this may become necessary soon enough.

**Comment by wusize** [comment]
The equivalence can be made by adding a third dimenson of zeros to the input tensors, e.g. [[1, 2, 0]], [[2, 1, 0]] and taking the last (3rd) dimension of the output (e.g. [[0, 0, -3]]).

**Comment by orenwang** [comment]
> "defaults to the first dimension found with the size 3"

whoa this one hit me hard today...
